Fix slow gradcheck when outputs that don't require grad precede those that do

soulitzer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #77667
* __->__ #77743
* 

Future work: fix for fast gradcheck as well, that would a bit require more effort though

Fixes: https://github.com/pytorch/pytorch/issues/77230

soulitzer committed:
Fix gradcheck when outputs that don't require grad precede those that do

[ghstack-poisoned]

soulitzer committed:
Update on "Fix gradcheck when outputs that don't require grad precede those that do"


Fixes: https://github.com/pytorch/pytorch/issues/77230

[ghstack-poisoned]

soulitzer committed:
Update on "Fix gradcheck when outputs that don't require grad precede those that do"


Fixes: https://github.com/pytorch/pytorch/issues/77230

[ghstack-poisoned]

Lezcano commented:
fwiw, to test that this works please uncomment the preprocessing formula used in the OpInfos of `linalg.lu` and in `linalg.slogdet`

soulitzer commented:
Ok its a bit more work to get this working with fast gradcheck, so just going to scope this PR to slow gradcheck. I think that's okay for now because we only use fast internally and we can just keep those wrappers for now.

soulitzer committed:
Update on "Fix slow gradcheck when outputs that don't require grad precede those that do"


Fixes: https://github.com/pytorch/pytorch/issues/77230

[ghstack-poisoned]

soulitzer committed:
Update on "Fix slow gradcheck when outputs that don't require grad precede those that do"

* 

Future work: fix for fast gradcheck as well, that would a bit require more effort though

Fixes: https://github.com/pytorch/pytorch/issues/77230

[ghstack-poisoned]
